pallets/crowdloans/src/lib.rs

```diff
@@ -959,6 +976,154 @@ pub mod pallet {
             }
             Ok(().into())
         }
+
+        #[pallet::weight(<T as Config>::WeightInfo::refund())]
+        #[transactional]
+        pub fn refund(origin: OriginFor<T>, crowdloan: ParaId) -> DispatchResult {
+
+            let who = ensure_signed(origin)?;
+            let mut pending_refunded_count = 0u32;
+            let mut contributed_refunded_count = 0u32;
+            let mut all_pending_refunded = false;
+            let mut all_contributed_refunded = false;
+
+            let vault = Self::current_vault(crowdloan).ok_or(Error::<T>::VaultDoesNotExist)?;
+            
+            // 1. check phase, should be Closed or Failed
+            ensure!(
+                vault.phase == VaultPhase::Closed || vault.phase == VaultPhase::Failed,
+                Error::<T>::IncorrectVaultPhase
+            );
+            
+            // 2. scan pending childstorage, mint RelayCurrency for contributor and
+            let (amount, _) = Self::contribution_get(
+                vault.trie_index, &who, ChildStorageKind::Pending
+            );
+            
+            ensure!(!amount.is_zero(), Error::<T>::NoContributions);
+
+            let pending_contributions = Self::contribution_iterator(
+                vault.trie_index, ChildStorageKind::Pending
+            );
+            for (who, (amount, _referral_code)) in pending_contributions {
```

Comment:
you can use `for (who, (amount, _referral_code)) in Self::contribution_iterator(
                vault.trie_index, ChildStorageKind::Pending
            )` to make things shorter